dashboard/app/views/levels/editors/fields/_code_area.html.haml

```diff
@@ -2,8 +2,9 @@
   Code Area
 
 #code_area.in.collapse
-  = render partial: 'levels/editors/droplet', locals: {f: f} if @level.uses_droplet?
-  = render partial: 'levels/editors/blockly', locals: {f: f} if !(@level.uses_droplet?) && @level.is_a?(Blockly)
+  = render partial: 'levels/editors/fields/droplet', locals: {f: f} if @level.uses_droplet?
+  = render partial: 'levels/editors/fields/blockly', locals: {f: f} if !(@level.uses_droplet?) && @level.is_a?(Blockly)
+  = render partial: 'levels/editors/fields/weblab_code_area', locals: {f: f} if @level.is_a?(Weblab)
   = render partial: 'levels/editors/fields/block_pools', locals: {f: f} if @level.is_a?(GamelabJr) || @level.is_a?(Dancelab)
   = render partial: 'levels/editors/fields/helper_libraries', locals: {f: f} if @level.is_a?(GamelabJr) || @level.is_a?(Dancelab)
-  = render partial: 'levels/editors/fields/validation_code', locals: {f: f}
+  = render partial: 'levels/editors/fields/validation_code', locals: {f: f} unless @level.is_a?(Weblab)
```

Comment:
I see just a few type checks here, which seems ok since it's the easiest way to keep the code area fields together even though the exact set of fields varies by level type. This is still a huge improvement over the huge number of type checks that existed beforehand!